### PR TITLE
Changed APIEndpoint.Port from int to int32

### LINKED ISSUE #1418: APIEndpoint.Port should be an int32
/kind feature

**Describe the solution you'd like**
To align with how other fields in Kubernetes and within this repository define ports, we should switch from a plain `int` to an `int32` in https://github.com/kubernetes-sigs/cluster-api/blob/ca112ae5f76c874c804a05e2ea2d2775d194cd39/api/v1alpha2/cluster_types.go#L142.

### PR DESCRIPTION
**Pull request by Xenwar**
Changed APIEndpoint.Port from int to int32

Fixes #1418

### CODE REVIEW

**Comment by detiber** [comment]
/hold
holding since this is a breaking API change, we should not merge this until we are ready to create a release branch for v1alpha2 related features and introduce breaking changes to master.

**Comment by ncdc** [comment]
v1alpha3 is starting to take shape!
/milestone v0.3.0

**Comment by vincepri** [comment]
@Xenwar Would you be able to rebase on master and make the change against the `v1alpha3` types?

**Comment by Xenwar** [comment]
> @Xenwar Would you be able to rebase on master and make the change against the `v1alpha3` types?

Done. Would please check.

File: api/v1alpha2/cluster_types.go [hunk 1]
```diff
@@ -139,7 +139,7 @@ type APIEndpoint struct {
 	Host string `json:"host"`
 
 	// The port on which the API server is serving.
-	Port int `json:"port"`
+	Port int32 `json:"port"`
```
Comment (vincepri) [inline api/v1alpha2/cluster_types.go]: This should stay `int`

Comment (Xenwar) [reply api/v1alpha2/cluster_types.go]: Done

**Comment by vincepri** [comment]
@Xenwar Seems you might need to run `make generate`

**Comment by Xenwar** [comment]
> @Xenwar Seems you might need to run `make generate`

No familiar with generate. 
of the following, which file do I need to add
```
	modified:   config/crd/bases/cluster.x-k8s.io_clusters.yaml
	modified:   go.mod

```

**Pull request merged**